ScreensaverHero.hx

```diff
@@ -0,0 +1,57 @@
+package;
+
+    import flixel.FlxSprite;
+    import flixel.util.FlxColor;
+    import flixel.FlxG;
+
+    class ScreensaverHero extends FlxSprite {
+        
+        public function new(x:Int = 0, y:Int = 0){
+            super(x,y);
+            super.makeGraphic(30, 30);
+        
+        }
+
+
+        override public function update(elapsed:Float){
+            
+            if (FlxG.keys.justPressed.SPACE){
+                this.velocity.x *= -1;
+                this.velocity.y *= -1;
+                }
+            
+            if (this.x <= (FlxG.width/2) && this.y <= (FlxG.height/2)){
+                this.color = 0xFF1500FF;
+            }
+            else if (this.x > (FlxG.width/2) && this.y <= (FlxG.height/2)){
+                this.color = 0xFFFFFB00;
+            }
+            else if (this.x <= (FlxG.width/2) && this.y > (FlxG.height/2)){
+                this.color = 0xFF16B900;
+            }
+            else if (this.x > (FlxG.width/2) && this.y > (FlxG.height/2)){
+                this.color = 0xFFFF0000;
+            }
+            
+
+            if (this.velocity.x > 0 && this.x >= FlxG.width - this.width){
+                this.x = 0;   
+            }
+            else if (this.velocity.x < 0 && this.x <= 0){
+                this.x = FlxG.width; 
+            }
+
+            if (this.velocity.y > 0 && this.y >= FlxG.height - this.height){
+                this.y = 0; 
+            }
+            else if (this.velocity.y < 0 && this.y <= 0){
+                this.y = FlxG.height;
+            }
```

Comment:
Nice work implementing screen-wrapping!

Couple of things to consider:
* Now that we've got three distinct kinds of behavior inside one `update` function (one block of code for changing movement directions, another for screen-wrapping the hero, another block of code for changing colors) it can be useful for us to break this up into a couple of different functions.

   Using this as an example, we could define methods called `screenWrap()`, `changeDirection()`, and `changeColor()`. `screenWrap` gets lines 37 - 49 inside of it, `changeDirection` gets lines 18 - 28 and `changeColor` gets lines 23 - 34. When those are off inside of their own separate functions, our update method gets to be super clean:

    ```
    override public function update(elapsed:Float){
        changeDirection();
        screenWrap();
        changeColor();
        super.update(elapsed);
    }
    ```

    Not a required change, but it might be worth practicing for when our update functions get more complex and difficult to handle.

* As a fun sort of exercise, how would you change this screen-wrapping logic to make the wrapping "smooth"? By that, I mean as the character fully disappears of one end of the screen, the edge of the player just starts to appear on the other end.

    Not a required change, but one that might be interesting to attempt.